src/main/java/com/alipay/oceanbase/rpc/util/ObHashUtils.java

```diff
@@ -118,7 +118,15 @@ public static long toHashcode(Object value, ObColumn refColumn, long hashCode,
         ObCollationType collationType = refColumn.getObCollationType();
 
         if (typeValue >= ObTinyIntType.getValue() && typeValue <= ObUInt64Type.getValue()) {
-            return ObHashUtils.longHash((Long) value, hashCode);
+            if (value instanceof Integer) {
+                return ObHashUtils.longHash(((Integer) value).longValue(), hashCode);
+            } else if (value instanceof Short) {
+                return ObHashUtils.longHash(((Short) value).longValue(), hashCode);
+            } else if (value instanceof Byte) {
+                return ObHashUtils.longHash(((Byte) value).longValue(), hashCode);
+            } else {
+                return ObHashUtils.longHash((Long) value, hashCode);
```

Comment:
handle boolean type

Reply:
There is no bool type in ObObj. We could use TinyInt (Byte not sure)  if we want to insert a boolean into a table with a bool type